src/sentry/tasks/auth.py

```diff
@@ -112,3 +137,46 @@ def _remove_2fa_non_compliant_member(member, org, actor=None, actor_key=None, ip
             context=email_context,
         )
         message.send_async([member.email])
+
+
+@instrumented_task(
+    name="sentry.tasks.remove_2fa_non_compliant_members",
+    queue="auth",
+    default_retry_delay=60 * 5,
+    max_retries=5,
```

Comment:
Are there any exceptions you'd like to catch and retry on? By default our instrumented_task decorator doesn't do any retries on its own.

Reply:
Celery can bind a [retry](https://docs.celeryproject.org/en/stable/userguide/tasks.html#retrying) method to a task that lets you capture exceptions and retry them manually. You can also use the [auto-retry](https://docs.celeryproject.org/en/stable/userguide/tasks.html#automatic-retry-for-known-exceptions) behavior.